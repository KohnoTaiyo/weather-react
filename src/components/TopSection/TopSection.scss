@use "../../styles/variables" as *;

h1 {
  text-align: center;
  font-size: $titleFontSize;
  margin-bottom: 16px;
}

h3 {
  margin-bottom: 8px;
}

form {
  margin-bottom: 16px;

  .error {
    color: $redColor;
    font-weight: bold;
    margin-top: 4px;
  }
}
