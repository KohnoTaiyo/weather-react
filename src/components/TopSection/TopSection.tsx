import "./TopSection.scss";

import { TextBox } from "../TextBox/TextBox";

export type TopSectionProps = {
  title: string;
  value: string;
  errorMessage?: string;
  onSubmit: (e: React.FormEvent<HTMLFormElement>) => void;
  onValueChange: (text: string) => void;
};

export function TopSection(props: TopSectionProps) {
  return (
    <>
      <h1>{props.title}</h1>
      <form onSubmit={props.onSubmit}>
        <TextBox value={props.value} onChange={props.onValueChange} />
        {props.errorMessage && <p className="error">{props.errorMessage}</p>}
      </form>
    </>
  );
}
