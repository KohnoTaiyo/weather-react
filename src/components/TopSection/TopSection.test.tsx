import { render, screen } from "@testing-library/react";

import { TopSection, TopSectionProps } from "./TopSection";

const defaultMock: TopSectionProps = {
  title: "title",
  value: "value",
  onValueChange: jest.fn(),
  onSubmit: jest.fn(),
};

describe("TopSection", () => {
  test("renders correctly dom", () => {
    const { container: noErrorMessageComponent } = render(<TopSection {...defaultMock} />);
    expect(screen.getByRole("heading", { level: 1 })).toHaveTextContent("title");
    expect(noErrorMessageComponent.querySelector(".error")).not.toBeInTheDocument();

    const { container: withErrorMessageComponent } = render(<TopSection {...defaultMock} errorMessage="error" />);
    expect(withErrorMessageComponent.querySelector(".error")).toBeInTheDocument();
  });

  test("snapshot", () => {
    const { container } = render(<TopSection {...defaultMock} errorMessage="error" />);
    expect(container).toMatchSnapshot();
  });
});
