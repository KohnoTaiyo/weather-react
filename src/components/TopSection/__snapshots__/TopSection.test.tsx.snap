// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopSection snapshot 1`] = `
<div>
  <h1>
    title
  </h1>
  <form>
    <div
      class="textBox"
    >
      <input
        class="textBox__input"
        placeholder="location"
        required=""
        type="text"
        value="value"
      />
      <input
        class="textBox__submit"
        tabindex="0"
        type="submit"
        value="Search"
      />
    </div>
    <p
      class="error"
    >
      error
    </p>
  </form>
</div>
`;
